### PR TITLE
[Persistence matrix] renaming matrix cells as matrix entries

### PR DESCRIPTION
**Pull request by hschreiber**
As we decided to rename "complex faces" as "complex cells", here a first PR renaming the matrix cells to avoid confusing both.

### CODE REVIEW

**Review by mglisse — APPROVED** [review]
(for obvious reasons I did not check the complete diff...)

File: src/Persistence_matrix/include/gudhi/Persistence_matrix/allocators/entry_constructors.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+ */
+
+/**
+ * @file entry_constructors.h
+ * @author Hannah Schreiber
+ * @brief Contains different versions of @ref Gudhi::persistence_matrix::Entry factories.
+ */
+
+#ifndef PM_COLUMN_CELL_CONSTRUCTORS_H
+#define PM_COLUMN_CELL_CONSTRUCTORS_H
```
Comment (VincentRouvreau) [inline src/Persistence_matrix/include/gudhi/Persistence_matrix/allocators/entry_constructors.h:18]: Maybe rename also the directives:
```suggestion
#ifndef PM_COLUMN_ENTRY_CONSTRUCTORS_H
#define PM_COLUMN_ENTRY_CONSTRUCTORS_H
```

File: src/Persistence_matrix/include/gudhi/Persistence_matrix/columns/entry_types.h [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+ * @author Hannah Schreiber
+ * @brief Contains the @ref Gudhi::persistence_matrix::Entry, @ref Gudhi::persistence_matrix::Entry_column_index and
+ * @ref Gudhi::persistence_matrix::Entry_field_element classes, as well as the
+ * @ref Gudhi::persistence_matrix::Dummy_entry_column_index_mixin and
+ * @ref Gudhi::persistence_matrix::Dummy_entry_field_element_mixin structures.
+ * Also defines the std::hash method for @ref Gudhi::persistence_matrix::Entry.
+ */
+
+#ifndef PM_MATRIX_CELL_H
+#define PM_MATRIX_CELL_H
```
Comment (VincentRouvreau) [inline src/Persistence_matrix/include/gudhi/Persistence_matrix/columns/entry_types.h:22]: Maybe rename also the directives:
```suggestion
#ifndef PM_MATRIX_ENTRY_H
#define PM_MATRIX_ENTRY_H
```

File: src/Persistence_matrix/include/gudhi/Persistence_matrix/allocators/entry_constructors.h (hunk 1 above)
Comment (hschreiber) [reply src/Persistence_matrix/include/gudhi/Persistence_matrix/allocators/entry_constructors.h]: Right, I forgot them! Should be done now.

**Review by VincentRouvreau — APPROVED** [review]

**Pull request merged**